.buildkite/steps/test-demo-notebooks.sh

```diff
@@ -53,6 +53,26 @@ exitCode=0
 # papermill will replace parameters on some notebooks to make them run faster in CI
 papermill --execution-timeout=600 --parameters_file "${stellargraph_dir}/.buildkite/notebook-parameters.yml" --log-output "$f" "$f" || exitCode=$?
 
-# and also upload the notebook with outputs, for someone to download and look at
-buildkite-agent artifact upload "$(basename "$f")"
+echo "+++ :jupyter: making result viewable"
+filename="$(basename "$f")"
+# and also upload the notebook with outputs, for someone to view by downloading or via nbviewer; we
+# can include a link to the latter automatically
+buildkite-agent artifact upload "$filename" 2>&1 | tee agent-output.txt
+
+artifact_id="$(sed -n "s/.*Uploading artifact \\(.*\\) $filename.*/\\1/p" agent-output.txt)"
```

Comment:
Would be be better to grep for the UUID format? Something like `grep -oE '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}'`

Reply:
That's a good idea. I went with something slightly more relaxed (just `[0-9a-f-]*`), but combined it with the filename check, because I thought it was good to match the `<UUID> <filename>` bit specifically, in case other UUIDs appear in the output for some reason (e.g. the job ID in a URL if it ends up having to retry something and reports a problem).